pandas/tseries/resample.py

```diff
@@ -1046,7 +1046,12 @@ def _get_binner_for_grouping(self, obj):
         l = []
         for key, group in grouper.get_iterator(self.ax):
             l.extend([key] * len(group))
-        grouper = binner.__class__(l, freq=binner.freq, name=binner.name)
+
```

Comment:
maybe isolate this a bit with some functions like _get_binner_for_resample does (*maybe* we should abstract this out even a bit more and have some PeriodIndexGrouper, DatetimeinexGrouper, which subclass TimeGrouper, but this might require some effort).